test/framework/integration_framework/fake_peer/behaviour/behaviour.hpp

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef INTEGRATION_FRAMEWORK_FAKE_PEER_BEHAVIOUR_HPP_
+#define INTEGRATION_FRAMEWORK_FAKE_PEER_BEHAVIOUR_HPP_
+
+#include <memory>
+#include <vector>
+
+#include "framework/integration_framework/fake_peer/fake_peer.hpp"
+#include "logger/logger.hpp"
+
+namespace integration_framework {
+  namespace fake_peer {
+
+    class Behaviour {
+     public:
+      virtual ~Behaviour();
+
+      /// Enable the behaviour for the given peer
+      void adopt(const std::shared_ptr<FakePeer> &fake_peer);
```

Comment:
Not agreed. From the interface point of view I would like to avoid copying.

Reply:
Couple of issues here:
1. First of all when you pass shared ptr by pointer, you are not guaranteed that it will live until the function ends. If can guarantee that from external code just pass a reference. If you do not have such guarantees pass it by value to ensure that.
2. Passing shared_ptr by value is cheap since its copy is very cheap, and you can always move that.
3. From interface point of view you do not need the fact that it is a pointer if you pass it by reference

Reply:
Argument no.1 together with the informal showing possession sharing by requiring a shared_ptr by value made me change my mind, thanks.